File: cracking/arrays/Ex11.py
def is_unique(input: str) -> bool:
    characters = list(input)
    characters.sort()
    for i in range(1, len(characters)):
        if characters[i] == characters[i - 1]:
            return False
    return True


assert is_unique("abcdefg")
assert not is_unique("abcda")
assert is_unique("")

File: cracking/arrays/Ex11.java
import java.util.Arrays;

public class Ex11 {

    static boolean isUnique(String input) {
        var characters = input.toCharArray();
        Arrays.sort(characters);
        for (int i = 1; i < characters.length; i++) {
            if (characters[i] == characters[i-1]) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        System.out.println(Ex11.isUnique("abdcefg"));
        System.out.println(Ex11.isUnique("abdcea"));
        System.out.println(Ex11.isUnique(""));
    }
}
